cdap-api-common/src/main/java/co/cask/cdap/api/macro/InvalidMacroException.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.macro;
+/**
+ * Indicates that there is an invalid macro.
+ */
+public class InvalidMacroException extends RuntimeException {
+
+  public InvalidMacroException(String message) {
+    super(message);
+  }
+
+  public InvalidMacroException(String message, Throwable cause) {
```

Comment:
i moved this from hydrator-plugins. its used in the case when macroFunction throws a runtime exception, but we want to propagate it as InvalidMacroException as it happened during macro evaluation.